profiles/tests/test_profiles_graphql_api.py

```diff
@@ -3049,6 +3053,162 @@ def test_database_stays_unmodified_when_mutation_is_not_completed(
+        permanent_foreign_address = (
+            verified_personal_information.permanent_foreign_address
+        )
+
+        expected_data = {
+            "myProfile": {
+                "verifiedPersonalInformation": {
+                    "firstName": verified_personal_information.first_name,
+                    "lastName": verified_personal_information.last_name,
+                    "givenName": verified_personal_information.given_name,
+                    "nationalIdentificationNumber": verified_personal_information.national_identification_number,
+                    "email": verified_personal_information.email,
+                    "municipalityOfResidence": verified_personal_information.municipality_of_residence,
+                    "municipalityOfResidenceNumber": verified_personal_information.municipality_of_residence_number,
+                    "permanentAddress": {
+                        "streetAddress": permanent_address.street_address,
+                        "postalCode": permanent_address.postal_code,
+                        "postOffice": permanent_address.post_office,
+                    },
+                    "temporaryAddress": {
+                        "streetAddress": temporary_address.street_address,
+                        "postalCode": temporary_address.postal_code,
+                        "postOffice": temporary_address.post_office,
+                    },
+                    "permanentForeignAddress": {
+                        "streetAddress": permanent_foreign_address.street_address,
+                        "additionalAddress": permanent_foreign_address.additional_address,
+                        "countryCode": permanent_foreign_address.country_code,
+                    },
+                },
+            }
+        }
+
+        executed = self._execute_query(rf, user_gql_client)
+
+        assert executed["data"] == expected_data
+
+    @pytest.mark.parametrize(
+        "address_type", ProfileWithVerifiedPersonalInformationTestBase.ADDRESS_TYPES,
+    )
+    def test_when_address_does_not_exist_returns_object_does_not_exist_error(
+        self, address_type, rf, user_gql_client
+    ):
+        profile = ProfileFactory(user=user_gql_client.user)
+        VerifiedPersonalInformationFactory(
+            profile=profile, **{address_type: None},
+        )
+
+        executed = self._execute_query(rf, user_gql_client)
+
+        assert_match_error_code(executed, "OBJECT_DOES_NOT_EXIST_ERROR")
+
+        received_info = executed["data"]["myProfile"]["verifiedPersonalInformation"]
+        for at in self.ADDRESS_TYPES:
+            received_address = received_info[to_graphql_name(at)]
+            if at == address_type:
+                assert received_address is None
+            else:
+                assert isinstance(received_address, dict)
+
+    @pytest.mark.parametrize("loa", ["substantial", "high"])
+    def test_high_enough_level_of_assurance_is_required(self, loa, rf, user_gql_client):
```

Comment:
Naming change: this doesn't test that `is_required` but that `is_allowed_access`
(This test together with the next one together verify `is_required`)

Reply:
Fixed in 7502c5b260.

Reply:
@tituomin Are you satisfied with the implemented changes?

Reply:
Well, now that you ask... To me, `is_needed` and `is_required` are synonyms. I would say `high_enough_level_of_access_gains_access`.

(Sorry to waste time on this... It's just that 'is_needed' to me still implies the implication (not high_enough) -> (no_access) , whereas my formulation does not.)

Reply:
No problem. Naming is important. Change in 70e26a4a5b.